cli/bpmetadata/tfconfig.go

```diff
@@ -395,3 +395,19 @@ func hasTfconfigErrors(diags tfconfig.Diagnostics) error {
 
 	return nil
 }
+
+// MergeExistingConnections merges existing connections from an old BlueprintInterface into a new one,
+// preserving manually authored connections.
+func MergeExistingConnections(newInterfaces, existingInterfaces *BlueprintInterface) {
```

Comment:
nit: since this is not used outside of this package, lets keep it unexported
```suggestion
func mergeExistingConnections(newInterfaces, existingInterfaces *BlueprintInterface) {
```